generators/chipyard/src/main/scala/CustomBusTopologies.scala

```diff
@@ -0,0 +1,100 @@
+
+package chipyard
+
+import freechips.rocketchip.config.{Field, Config, Parameters}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util.{Location, Symmetric}
+import freechips.rocketchip.subsystem._
+
+// I'm putting this code here temporarily as I think it should be a candidate
+// for upstreaming based on input from Henry Cook, but don't wnat to deal with
+// an RC branch just yet.
```

Comment:
Yup, I would encourage you to upstream this. Though I am thinking it might be best to just bite the bullet of deprecating my old `SubsystemCrossingParams` in favor of the pattern you used here for `sbusToMbusXType` (i.e. make upstream `HierarchicalBusTopologyParams` take three `ClockCrossingType` arguments with no extra case class layer to have to juggle alterations upon.) If you provide the mix-ins too I will get our stuff ported to using them.

Reply:
I'm happy to just enumerate the conventional crossings and give them names in the config mix-ins. 

It would be cool if, in `p` or elsewhere, we had a Map[(Location, Location), ClockCrossingType] or something to that effect.